### PR TITLE
fix(i18n): handle unicode characters properly in Input

### PR DESCRIPTION
**Pull request by fanantoxa**
Updates input component to handle Unicode characters properly
- [x] Ctrl + p
- [x] Ctral + Shft + p
- [x] `:`, `/`
- [x] Ctrl + SHft + f
- [x] SCM
- [x] Add tests

### CODE REVIEW

**Comment by fanantoxa** [comment]
@glennsl I have one bug with this implementation. Not sure if this should be fixed here or somewhere else:
So the problem is with `:` and `/` Menus.
When I start put Unicode (Ukrainian in this case) characters in at it add them properly and updates cursor properly.
If I press `<BS>` at the end it removes last character properly.
But If I press `Left` It doesn't update it to proper cursor position
(I have feeling that it still count 8byte chars in this case)
Could you help me with this please?

comand_bug

**Comment by bryphe** [comment]
This is awesome @fanantoxa , thank you for improving this!

> But If I press Left It doesn't update it to proper cursor position
(I have feeling that it still count 8byte chars in this case)
Could you help me with this please?

Ah, interesting... I think this is similar to https://github.com/onivim/oni2/pull/1608 - where the 'cursor position' we're getting from Vim is being reported as the _byte position_ instead of a _character index_ - so we have to convert it from byte -> character to get the proper 'cursor caret position'. I'll take a look a bit more tomorrow

**Comment by fanantoxa** [comment]
@glennsl Hey It's ready for review :wink:

**Comment by fanantoxa** [comment]
@bryphe I was told to ask you for review)

File: src/Feature/InputText/Model.re [hunk 1]
```diff
@@ -288,11 +278,13 @@ let%test_module "Model" =
...
 
+     let uTestString = "😊↪Вім is Cool";
+
      let%test_module "paste" =
        (module
         {
-          let pasteText = "hello from clipboard";
-          let pasteTextLength = pasteText |> String.length;
+          let pasteText = "😊↪Вім hello from clipboard";
+          let pasteTextLength = pasteText |> Zed_utf8.length;
```
Comment (bryphe) [inline src/Feature/InputText/Model.re:287]: Nice! Thanks for adding all these test cases @fanantoxa 😄

**Comment by bryphe** [comment]
Thanks for the ping @fanantoxa ! Changes look great to me. Really appreciate all the test cases you've updated & added.

**Pull request merged**